File: src/main/resources/output/IFC4X3_java/Entities/IfcRoot/IfcObjectDefinition/IfcObject/IfcProduct/IfcElement/IfcDistributionElement/IfcDistributionFlowElement/IfcEnergyConversionDevice/IfcBurner/IfcBurner.java
public class IfcBurner extends IfcEnergyConversionDevice {
    public IfcBurnerTypeEnum PredefinedType;

    // === WHERE CLAUSES ===
    // CorrectPredefinedType : NOT(EXISTS(PredefinedType)) OR (PredefinedType <> IfcBurnerTypeEnum.USERDEFINED) OR ((PredefinedType = IfcBurnerTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcObject.ObjectType))
    // CorrectTypeAssigned : (SIZEOF(IsTypedBy) = 0) OR ('IFC4X3_DEV_73740fe4.IFCBURNERTYPE' IN TYPEOF(SELF\IfcObject.IsTypedBy[1].RelatingType))
}

File: src/main/resources/output/IFC4X3_vbnet/Entities/IfcRoot/IfcObjectDefinition/IfcObject/IfcProduct/IfcElement/IfcDistributionElement/IfcDistributionFlowElement/IfcEnergyConversionDevice/IfcBurner/IfcBurner.vb
Public Class IfcBurner Inherits IfcEnergyConversionDevice
    Public PredefinedType As IfcBurnerTypeEnum

    ' === WHERE CLAUSES ===
    ' CorrectPredefinedType : NOT(EXISTS(PredefinedType)) OR (PredefinedType <> IfcBurnerTypeEnum.USERDEFINED) OR ((PredefinedType = IfcBurnerTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcObject.ObjectType))
    ' CorrectTypeAssigned : (SIZEOF(IsTypedBy) = 0) OR ('IFC4X3_DEV_73740fe4.IFCBURNERTYPE' IN TYPEOF(SELF\IfcObject.IsTypedBy[1].RelatingType))
End Class
